examples/gltf/app.js

```diff
@@ -203,22 +215,32 @@ export class DemoApp {
   }
 
   onRender({gl, time, width, height, aspect}) {
-    gl.viewport(0, 0, width, height);
     clear(gl, {color: [0.2, 0.2, 0.2, 1.0], depth: true});
 
+    this.vrDisplay.getFrameData(this.frameData);
+
+    gl.viewport(0, 0, width * 0.5, height);
+    this.onRenderEye({gl, time}, 'left');
+
+    gl.viewport(width * 0.5, 0, width * 0.5, height);
+    this.onRenderEye({gl, time}, 'right');
+  }
+
+  onRenderEye({gl, time}, eye) {
     const [pitch, roll] = this.rotation;
     const cameraPos = [
       -this.translate * Math.sin(roll) * Math.cos(-pitch),
       -this.translate * Math.sin(-pitch),
       this.translate * Math.cos(roll) * Math.cos(-pitch)
     ];
 
-    const uView = new Matrix4()
-      .translate([0, 0, -this.translate])
-      .rotateX(pitch)
-      .rotateY(roll);
+    const uView = new Matrix4(Array.from(this.frameData[`${eye}ViewMatrix`]));
```

Comment:
It looks like the frameData matrices are already typed arrays: https://developer.mozilla.org/en-US/docs/Web/API/VRFrameData/leftViewMatrix

So you can get rid of the copies you're making here.

Reply:
If I don't make copies the whole browser crashes! Not sure why.

Reply:
Wow, that sounds like a serious browser bug. You should file a report, [here](https://bugs.chromium.org/p/chromium/issues/list) for Chrome or [here](https://bugzilla.mozilla.org/) for Firefox. There are security implications to being able to crash the browser from JS, so they'll definitely want to hear about it.

Reply:
It did not crash just became super-slow. Suspect it's a memory leak.
I'm using an extension for WebVR simulation, so it could be the extension and not the browser.

Reply:
Please add a TODO comment in the code explaining why we leave these in. We need to go back and investigate the situation later.